src/Nats/Connection.php

```diff
@@ -239,16 +239,16 @@ public function connect($timeout = null)
         $msg = 'CONNECT '.$this->options;
         $this->send($msg);
         $connect_response = $this->receive();
-        if (strpos($connect_response, '-ERR')!== false) {
-            throw new \Exception("Failing connection: $connect_response");
+
+        if ($this->isErrorResponse($connect_response)) {
+            throw Exception::forFailedConnection($connect_response);
         }
 
         $this->ping();
         $ping_response = $this->receive();
-        if ($ping_response !== "PONG") {
```

Comment:
I removed this as we could just check for `-ERR` and get the expected behaviour.